### PR TITLE
Feature 3.1

### PR DESCRIPTION
**Pull request by jasonyang1992**

### CODE REVIEW

**Review by andrewvoroshylo — APPROVED** [review]

**Review by DJSHS — APPROVED** [review]

**Review by pnguye17 — APPROVED** [review]

File: src/app/components/admin/admin.component.ts [hunk 1]
```diff
@@ -13,25 +15,59 @@ import { Router } from '@angular/router';
...
+   searchText;
   ngOnInit() {
+
+    this.adminservice.showAllUser()
+    .subscribe(
+      data=> {
+        this.listofUsers = data;
+      }
+    )
+
```
Comment (carolynrehm) [inline src/app/components/admin/admin.component.ts]: @jasonyang1992  this functionality that is already found in the user service; can we consolidate this implementation into one service?

File: src/app/components/admin/admin.component.ts [hunk 2]
```diff
@@ -13,25 +15,59 @@ import { Router } from '@angular/router';
...
+  }
+
+  private refreshData(){
+    this.adminservice.showAllUser()
+    .subscribe(
+      data=> {
+        this.listofUsers = data;
+      }
+    )
+  }
```
Comment (carolynrehm) [inline src/app/components/admin/admin.component.ts]: can we display this information without making an entirely new http call?

File: src/app/pipes/adminfilter.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  name: 'adminfilter'
+})
+export class AdminfilterPipe implements PipeTransform {
+
+  transform(items: any[], searchText: Object): any {
+    if (!items || !searchText) {
+           return items;
+       }
+     return items.filter(item => item.userName.indexOf(searchText) !== -1);
+  }
```
Comment (carolynrehm) [inline src/app/pipes/adminfilter.pipe.ts]: Angular advises against using pipes for searching and sorting ([see documentation here](https://angular.io/guide/pipes#appendix-no-filterpipe-or-orderbypipe)) - can we do this using a service method instead?

**Review by carolynrehm — CHANGES_REQUESTED** [review]
see comments

**Review by carolynrehm — CHANGES_REQUESTED** [review]
@jasonyang1992 resolve merge conflicts

**Review by carolynrehm — COMMENTED** [review]
Also, if possible remove changes from package.json and app.module.ts because they're not relevant. Tag me when you're ready for review

File: src/app/components/admin/admin.component.ts [hunk 4]
```diff
@@ -13,25 +15,47 @@ import { Router } from '@angular/router';
...
     sessionStorage.clear();
     this.router.navigate(['']);
   }
+
+  searchUser(){
+    this.listofUsers = this.allListofUsers.filter(item => item.userName.toLowerCase().includes(this.searchText.toLowerCase()));
+  }
+
+  banning(obj: User, userid: number, userName: string, firstName: string, lastName: string, email: string, phoneNumber:string, batch: object, active: boolean) {
+    console.log(obj);
```
Comment (carolynrehm) [inline src/app/components/admin/admin.component.ts]: @jasonyang1992 
- remove console log here 
- is there a reason for passing all of these parameters in along with the user object? wouldn't all of those fields already be a part of the user object?

File: src/app/components/admin/admin.component.html [hunk 5]
```diff
@@ -1,3 +1,43 @@
...
+    </tr>
+
+    <tr *ngFor="let get of listofUsers">
+      <td>{{get.userId}}</td>
+      <td>{{get.userName}}</td>
+      <td>{{get.firstName}}</td>
+      <td>{{get.lastName}}</td>
+      <td>{{get.email}}</td>
+      <td>{{get.phoneNumber}}</td>
+      <td *ngIf="get.batch">{{get.batch.batchNumber}}</td>
```
Comment (carolynrehm) [inline src/app/components/admin/admin.component.html]: This ngIf makes things render inconsistently if there is no batch; puts "isActive" value in the batch column and the ban button in the "isActive" column

File: src/app/components/admin/admin.component.html [hunk 6]
```diff
@@ -1,3 +1,43 @@
...
+      <th> First Name </th>
+      <th> Last Name </th>
+      <th> Email </th>
+      <th> Phone Number </th>
+      <th> Batch Number</th>
+      <th> is Active </th>
+      <th> Ban </th>
+    </tr>
+
+    <tr *ngFor="let get of listofUsers">
```
Comment (carolynrehm) [inline src/app/components/admin/admin.component.html]: can we change this naming to something a little more intuitive? perhaps "let user of users" having users represent the list of users?

File: src/app/components/admin/admin.component.html (hunk 5 above)
Comment (carolynrehm) [reply src/app/components/admin/admin.component.html]: we can use a ternary here to check if the batch exists and assigning the string interpolated value to the batch number if it does, or "n/a" if it doesnt

**Review by carolynrehm — APPROVED** [review]

**Comment by carolynrehm** [comment]
@jasonyang1992 merge conflict needs resolution before I can merge

**Review by carolynrehm — APPROVED** [review]

**Pull request merged**